docs/developers/architecture/app_model.md

```diff
@@ -2,66 +2,195 @@
+singleton `app`:
+
+```python
+from napari._app_model import get_app
+
+
+get_app().register_action(action)
+```
+
+````{note}
+If you're following along in the console, you may see the following error
+when executing the above code:
+
+```python
+ValueError: Command 'napari:layer:split_rgb' already registered
+```
+
+This is because command id's may currently only be registered once, and associated with a single callback (and napari's internal app already used the `CommandId.LAYER_SPLIT_RGB` id).
+````
+
+(app-model-actions-napari)=
+
+### `Action`s in napari
+
+In napari non-Qt `Action`s are defined in
+[`napari/_app_model/actions`](https://github.com/napari/napari/tree/main/napari/_app_model/actions)
+while Qt `Action`s are defined in
+[`napari/_qt/_qapp_model/qactions`](https://github.com/napari/napari/tree/main/napari/_qt/_qapp_model/qactions).
+Non-Qt `Action`s get registered with `app` during
+initialization of the napari `app`, in `NapariApplication`'s
+{meth}`~napari._app_model._app.NapariApplication.__init__`. Qt `Action`s
+get registered in
+{func}`~napari._qt._qapp_model.qactions.init_qactions`, which gets
+called during initialization of `_QtMainWindow`. Note this is relevant when
+considering the differences between app-model and action manager implementation,
+see [](app-model-action-manager-differences) for more.
+
+(app-model-commands)=
 
 ## Commands
 
 Commands represent callable objects that "do something" in napari, and usually
 have a corresponding representation somewhere in the GUI (such as a button, a
 menu item, or a keybinding).
 
-All commands have a string id (e.g. '`napari:layer:duplicate'`), which should be
-declared as a member of the `napari._app_model.constants.CommandId` enum.  Internally,
-an instance of this enum should be used instead of the string literal when
-referring to a command, as it is easier to refactor and test.
-
-```{note}
-Some of these command strings MAY be exposed externally in the future. For example, a plugin may wish to refer to a napari command.
-```
+Inline (lambda) and nested functions should be avoided for command callbacks to
+prevent memory leakage.
+
+All commands have a string id (e.g. '`napari:layer:duplicate`'). These are
+currently `CommandID` enums but we are currently considering changing to using
+plain strings to simplify decalaration of `Action`s.
+
+{class}`app_model.types.CommandRule` class (of which {class}`app_model.types.Action`
+is a subclass) includes many fields to detail various aspects of commands.
+See {class}`app_model.types.CommandRule` for details on every field.
+Notable {class}`~app_model.types.CommandRule` fields
+are `enablement` and `toggled`, which both can take an
+{class}`app_model.expressions.Expr`, an expression that can be evaluated, with
+context if required (see [](context-expressions) for more details).
+`enablement` determines whether the command is enabled in the UI. It does not prevent
```

Comment:
```suggestion
`enablement` determines whether the command is visible in the UI. It does not prevent
```

Reply:
Nah `when` controls visibility. A visible, disabled action appears gray-ed out on the menu.